File: 2022/python/d1-gpt.py
# solution written by GPT chatbot
# works and provides correct results

# parse the input
calories = []
current_elf = []
with open('../inputs/day1') as f:
    for line in f:
        # if the line is blank, store the current elf's calories and start a new elf
        if line.strip() == "":
            calories.append(sum(current_elf))
            current_elf = []
        else:
            # add the calorie count to the current elf's list
            current_elf.append(int(line))

# add the last elf's calories to the list
calories.append(sum(current_elf))

# find the elf with the most calories
most_calories = max(calories)

# print the result T1
print(f"The elf carrying the most calories has a total of {most_calories} calories.")

# find the top three elves with the most calories
calories.sort(reverse=True)
top_three_calories = calories[:3]

# calculate the total number of calories for the top three elves
total_calories = sum(top_three_calories)

# print the result T2
print(f"The top three elves are carrying a total of {total_calories} calories.")

File: 2022/elixir/d1-gpt.exs
# parse the input
calories =
  File.stream!("../inputs/day1")
  |> Stream.map(&String.trim/1)
  |> Stream.chunk_by(&(&1 == ""))
  |> Stream.reject(&(&1 == [""]))
  |> Enum.map(fn chunk ->
    chunk
    |> Enum.map(&String.to_integer/1)
    |> Enum.sum()
  end)

# find the elf with the most calories
most_calories = Enum.max(calories)

# print the result
IO.puts("The elf carrying the most calories has a total of #{most_calories} calories.")

# find the top three elves with the most calories
top_three_calories =
  Enum.sort(calories, &(&1 >= &2))
  |> Enum.take(3)

# calculate the total number of calories for the top three elves
total_calories = Enum.reduce(top_three_calories, 0, &(&1 + &2))

# print the result
IO.puts("The top three elves are carrying a total of #{total_calories} calories.")
